packages/shared/src/utils/node/ensure-extension-dirs.test.ts

```diff
@@ -0,0 +1,16 @@
+import { ensureExtensionDirs } from '.';
+import { EXTENSION_TYPES } from '../../constants/extensions';
+import { ExtensionType } from '../../types';
+
+describe('ensureExtensionDirs', () => {
+	const types = EXTENSION_TYPES as readonly ExtensionType[];
+	it('returns undefined if the folders exist', async () => {
+		expect(await ensureExtensionDirs('./extension', types)).toBe(undefined);
+	});
+
+	it('throws an error when a folder cant be opened', () => {
```

Comment:
We should add some before/after functions here to setup/teardown the folder structure to test, to ensure other tests/repeated invocations don't cause state-based inconsistencies in the test results